samples/Basic-Bot/LanguageGenerationUtilities.cs

```diff
@@ -16,33 +16,21 @@ public static class LanguageGenerationUtilities
         /// Creates an instance of <see cref="LanguageGenerationResolver"/>.
         /// </summary>
         /// <param name="applicationId">Language generation application id.</param>
-        /// <param name="endpointKey">The language generation service subscription key.</param>
-        /// <param name="endpointRegion">The language generation region.</param>
+        /// <param name="applicationRegion">Language generation application region.</param>
+        /// <param name="applicationLocale">Language generation application locale.</param>
+        /// <param name="applicationVersion">Language generation application version.</param>
+        /// <param name="subscriptionKey">The language generation service subscription key.</param>
         /// <returns>An instance of the language generation resolver.</returns>
-        public static LanguageGenerationResolver CreateResolver(string applicationId, string endpointKey, string endpointRegion)
+        public static LanguageGenerationResolver CreateResolver(string applicationId, string applicationRegion, string applicationLocale, string applicationVersion, string subscriptionKey)
         {
-            if (string.IsNullOrEmpty(applicationId))
-            {
-                throw new ArgumentException("Application id can't be null or empty.", nameof(applicationId));
-            }
-
-            if (string.IsNullOrEmpty(endpointKey))
-            {
-                throw new ArgumentException("Endpoint subscription key can't be null or empty.", nameof(applicationId));
-            }
-
-            if (string.IsNullOrEmpty(endpointRegion))
-            {
-                // Set default region to westus
-                endpointRegion = "westus";
-            }
 
-            var languageGenerationEndpoint = $"https://{endpointRegion}.cts.speech.microsoft.com/v1/lg";
-            var tokenIssuingEndpoint = $"https://{endpointRegion}.api.cognitive.microsoft.com/sts/v1.0/issueToken";
+            var application = new LanguageGenerationApplication(applicationId, applicationRegion, applicationLocale, applicationVersion, subscriptionKey);
 
-            var application = new LanguageGenerationApplication(applicationId, endpointKey, languageGenerationEndpoint);
+            var languageGenerationEndpoint = $"https://{applicationRegion}.cts.speech.microsoft.com/v1/lg";
```

Comment:
Why do we have this hard coded in multiple regions?

Reply:
This is only placed in two places, here to demonstrate the ability to override the endpoints through the LanguageGenerationOptions object, and in the DefaultRegionalEndpointProvider for the default endpoint provider.